src/org/traccar/protocol/FlespiProtocolDecoder.java

```diff
@@ -69,36 +85,246 @@ protected Object decode(
     }
 
     private void decodePosition(JsonObject object, Position position) {
-        position.setProtocol(getProtocolName());
+        // store all params in position
+        Set<Map.Entry<String, JsonValue>> params = object.entrySet();
+        Iterator<Map.Entry<String, JsonValue>> it = params.iterator();
+        while (it.hasNext()) {
+            Map.Entry<String, JsonValue> param = it.next();
+            String paramName = param.getKey();
+            JsonValue paramValue = param.getValue();
+            if ("timestamp".equals(paramName)) {
```

Comment:
Why do we need split it into several methods? Why can't we just have a big switch for everything?

Reply:
Having one big switch will violate MethodLength checkstyle rule. Is it possible to discard this check?

Reply:
Are you sure you have tried switch and not ifs?